crates/router/src/core/payment_link.rs

```diff
@@ -267,16 +254,155 @@ fn validate_order_details(
     Ok(updated_order_details)
 }
 
-fn extract_payment_link_config(
-    pl_config: Option<serde_json::Value>,
-) -> Result<Option<admin_types::PaymentLinkConfig>, error_stack::Report<errors::ApiErrorResponse>> {
-    pl_config
-        .map(|config| {
-            serde_json::from_value::<admin_types::PaymentLinkConfig>(config)
-                .into_report()
-                .change_context(errors::ApiErrorResponse::InvalidDataValue {
-                    field_name: "payment_link_config",
-                })
+pub fn extract_business_payment_link_config(
+    pl_config: serde_json::Value,
+) -> Result<admin_types::BusinessPaymentLinkConfig, error_stack::Report<errors::ApiErrorResponse>> {
+    serde_json::from_value::<admin_types::BusinessPaymentLinkConfig>(pl_config.clone())
+        .into_report()
+        .change_context(errors::ApiErrorResponse::InvalidDataValue {
+            field_name: "payment_link_config",
+        })
+}
+
+pub fn extract_payment_link_config(
+    pl_config: serde_json::Value,
+) -> Result<
+    admin_types::PaymentCreatePaymentLinkConfig,
+    error_stack::Report<errors::ApiErrorResponse>,
+> {
+    serde_json::from_value::<admin_types::PaymentCreatePaymentLinkConfig>(pl_config.clone())
+        .into_report()
+        .change_context(errors::ApiErrorResponse::InvalidDataValue {
+            field_name: "payment_link_config",
         })
-        .transpose()
+}
+
+pub fn get_payment_link_config_based_on_priority(
+    payment_create_link_config: Option<admin_types::PaymentCreatePaymentLinkConfig>,
+    business_link_config: Option<serde_json::Value>,
+    merchant_name: String,
+    default_domain_name: String,
+) -> Result<
+    (admin_types::PaymentCreatePaymentLinkConfig, String),
+    error_stack::Report<errors::ApiErrorResponse>,
+> {
+    match (payment_create_link_config, business_link_config) {
+        (Some(payment_create), Some(business)) => {
```

Comment:
yes, but this function doesn't just check the object the field inside the object being passed, since all the fields inside are optional.

example.

let's say a merchant has provided logo in business level payment link config

whereas in payment create it just wants to change the theme, so ideal behaviour would be to take logo from business level and theme from payment create